From 4bb79120c88c77d605d45c94cec4fa0f29ac957c Mon Sep 17 00:00:00 2001
From: Gu, Wangyi <wangyi.gu@intel.com>
Date: Wed, 4 Sep 2013 21:07:53 +0800
Subject: improve the UX when input clip with implicit SBR signaling AAC(1)

BZ: 132381 126865

this patch let the implicit SBR signaling AAC do the transcoding.

Category:aosp improvement
Domain: Video.Media-video editor
Origin: internal
Upstream-Candidate: No

Change-Id: Ia01873829d1b354c6d84262cf5049d66ba09d84c
Orig-Change-Id: Iaf26eaef6d09d7a049b5b74f1cf439e0d9cebda3
Signed-off-by: Gu, Wangyi <wangyi.gu@intel.com>
Reviewed-on: http://android.intel.com:8080/129893
Reviewed-by: Feng, Wei <wei.feng@intel.com>
Reviewed-by: Shi, PingX <pingx.shi@intel.com>
Tested-by: Shi, PingX <pingx.shi@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 libvideoeditor/vss/mcs/src/M4MCS_API.c |   14 +++++++++++---
 libvideoeditor/vss/src/M4xVSS_API.c    |    3 +++
 2 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/libvideoeditor/vss/mcs/src/M4MCS_API.c b/libvideoeditor/vss/mcs/src/M4MCS_API.c
index 42717f9..b22795b 100755
--- a/libvideoeditor/vss/mcs/src/M4MCS_API.c
+++ b/libvideoeditor/vss/mcs/src/M4MCS_API.c
@@ -9793,14 +9793,19 @@ static M4OSA_ERR M4MCS_intGetInputClipProperties( M4MCS_InternalContext *pC )
             pC->InputFileProperties.uiSamplingFrequency =
                 pC->AacProperties.aSampFreq;
 
-            if( pC->AacProperties.aSBRPresent )
+            if( pC->AacProperties.aSBRPresent == 1)
             {
                 pC->InputFileProperties.AudioStreamType =
                     M4VIDEOEDITING_kAACplus;
                 pC->InputFileProperties.uiExtendedSamplingFrequency =
                     pC->AacProperties.aExtensionSampFreq;
             }
-
+#ifdef VIDEOEDITOR_INTEL_NV12_VERSION
+            if (pC->AacProperties.aSBRPresent == -1){
+                pC->InputFileProperties.uiExtendedSamplingFrequency =
+                   pC->AacProperties.aSampFreq;
+            }
+#endif
             if( pC->AacProperties.aPSPresent )
             {
                 pC->InputFileProperties.AudioStreamType =
@@ -10667,7 +10672,10 @@ M4OSA_ERR M4MCS_intCheckAndGetCodecProperties(
                 pC->pReaderAudioStream->m_nbChannels;
             pC->AacProperties.aSampFreq =
                 pC->pReaderAudioStream->m_samplingFrequency;
-
+#ifdef VIDEOEDITOR_INTEL_NV12_VERSION
+            if (pC->AacProperties.aSBRPresent == 0)
+                pC->AacProperties.aSBRPresent = -1;
+#endif
         }
 
     } else if( err != M4NO_ERROR) {
diff --git a/libvideoeditor/vss/src/M4xVSS_API.c b/libvideoeditor/vss/src/M4xVSS_API.c
index 9f5410b..a5b7c64 100755
--- a/libvideoeditor/vss/src/M4xVSS_API.c
+++ b/libvideoeditor/vss/src/M4xVSS_API.c
@@ -2412,6 +2412,9 @@ M4OSA_ERR M4xVSS_SendCommand( M4OSA_Context pContext,
                 != xVSS_context->pSettings->xVSS.outputAudioFormat
                 || (fileProperties.AudioStreamType == M4VIDEOEDITING_kAAC
                 && (fileProperties.uiSamplingFrequency != samplingFreq
+#ifdef VIDEOEDITOR_INTEL_NV12_VERSION
+                || fileProperties.uiExtendedSamplingFrequency > 0
+#endif
                 || bAudioMono
                 != xVSS_context->pSettings->xVSS.bAudioMono)) )
             {
-- 
1.7.4.1

